http/query_handler.go

```diff
@@ -520,14 +520,12 @@ func (s FluxQueryService) Check(ctx context.Context) check.Response {
 }
 
 // GetQueryResponse runs a flux query with common parameters and returns the response from the query service.
-func GetQueryResponse(qr *QueryRequest, addr, org, token string, headers ...string) (*http.Response, error) {
+func GetQueryResponse(qr *QueryRequest, addr *url.URL, org, token string, headers ...string) (*http.Response, error) {
 	if len(headers)%2 != 0 {
 		return nil, fmt.Errorf("headers must be key value pairs")
 	}
-	u, err := NewURL(addr, prefixQuery)
-	if err != nil {
-		return nil, err
-	}
+	u := *addr
+	u.Path = prefixQuery
```

Comment:
Is modifying the URL safe here? Does the caller expect that?

Reply:
AFAIK it should be safe because it's a copy of `addr`, the value passed in from the caller.